frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx

```diff
@@ -72,17 +92,22 @@ const mapDispatchToProps = {
   setParameterMapping,
 };
 
-DashCardCardParameterMapper.propTypes = {
-  card: PropTypes.object.isRequired,
-  dashcard: PropTypes.object.isRequired,
-  editingParameter: PropTypes.object.isRequired,
-  target: PropTypes.array,
-  mappingOptions: PropTypes.array.isRequired,
-  metadata: PropTypes.object.isRequired,
-  setParameterMapping: PropTypes.func.isRequired,
-  isMobile: PropTypes.bool,
-  question: PropTypes.object,
-};
+interface DashcardCardParameterMapperProps {
+  card: Card;
+  dashcard: DashboardCard;
+  editingParameter: Parameter | null | undefined;
+  target: ParameterTarget | null | undefined;
+  setParameterMapping: (
+    parameterId: ParameterId,
+    dashcardId: DashCardId,
+    cardId: CardId,
+    target: ParameterTarget | null,
+  ) => void;
+  isMobile: boolean;
+  // virtual cards will not have question
+  question?: Question;
+  mappingOptions: any[];
```

Comment:
any chance we could avoid this `any`?

Reply:
oh, right 😄 will do

Reply:
[1421ca9](https://github.com/metabase/metabase/pull/38596/commits/1421ca98b1869665984e6ad9a0a7571b539bd3ac)